lib/core/utils/is-hidden.js

```diff
@@ -35,6 +40,11 @@ axe.utils.isHidden = function isHidden(el, recursed) {
 	}
 
 	parent = el.assignedSlot ? el.assignedSlot : el.parentNode;
+	isHidden = axe.utils.isHidden(parent, true);
+
+	if (node) {
+		node.isHidden = isHidden;
```

Comment:
I don't really like that `virtualNode.isHidden` becomes unpredictable. At any point in a run, that property may get assigned, but it's hard to know when. We either have to make this prop not accessible at all, or make sure it's always available. The way I originally thought this might work is by using a getter method for `virtualNode.isHidden`. Let's discuss.

Reply:
I had that thought too.

However, if we define the _entire_ `VirtualNode` interface when we create it and default `isHidden` to `null` (or something?), we'll always know if it's available and what properties to expect on the `VirtualNode`.

Reply:
The problem with a getter on the virtual node is that it isn't a call that just looks at the node in isolation, it has to look up the entire parent tree. It's best to cache the results as you go so you only look up the the parent tree once and not for each node that needs it (like we did before)

Reply:
> I had that thought too.
> 
> However, if we define the _entire_ `VirtualNode` interface when we create it and default `isHidden` to `null` (or something?), we'll always know if it's available and what properties to expect on the `VirtualNode`.

That would be fine to me